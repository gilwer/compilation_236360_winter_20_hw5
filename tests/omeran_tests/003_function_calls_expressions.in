// calling functions as expressions
enum days { SUN, MON, TUES };

int foo(int a) {
  return a;
}

int kaki(int a, int bb, int c) {
  int x = a + bb + c;
  return x*2;
}

void bar() {
  return;
}

void baz() {
  int t;
}

void main() {
  int x = 3;
  printi(x); // 3
  x = kaki(foo(3), x, 4);
  printi(x); // 20
  int y = foo(2);
  printi(y); // 2
  int z = foo(x);
  printi(z); // 20
  z = foo(44+33);
  printi(z); // 77
  y = foo (x * 7);
  printi(y); // 140
  bar();
  baz();
}