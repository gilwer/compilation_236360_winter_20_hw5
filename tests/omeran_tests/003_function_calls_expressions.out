3
20
2
20
77
140
